### ISSUE TITLE
Issue formatting 1 hour at 23.976

### ISSUE THREAD

**Issue by sroucheray** [CrystalComputerCorp/smpte-timecode#37]
Hi,

Thanks for your work !

While doing a bunch of tests I encounter an issue. I have an array of duration in seconds with associated frame rates. I format them and check the expected formatted timecode output. 

In the excerpt below, for each of the cases, to obtain the frames count parameter of the `Timecode` function I do `duration x framerate`, obviously. The `timecode` field is the actual result (not what I expected).

You can see that for the frame rate `23.976`, the actual result is not `01:00:00:00` but `00:59:56:10`. Is that a bug, do I miss something specific to this frame rate, or maybe my test is not well built ?

```json
[
  { duration: 3600, framerate: 23.976, timecode: '00:59:56:10' },
  { duration: 3600, framerate: 24, timecode: '01:00:00:00' },
  { duration: 3600, framerate: 25, timecode: '01:00:00:00' },
  { duration: 3600, framerate: 29.97, timecode: '01:00:00;00' },
  { duration: 3600, framerate: 30, timecode: '01:00:00:00' },
  { duration: 3600, framerate: 50, timecode: '01:00:00:00' },
  { duration: 3600, framerate: 59.94, timecode: '01:00:00;00' },
  { duration: 3600, framerate: 60, timecode: '01:00:00:00' }
]
```

**Comment by m1tk4** [comment]
I'm not sure I fully understand how the comparison is done (for example, how do you instantiate the Timecode obj) - maybe you could provide that.

There was a recent bugfix for 23.976 that fixed the bug with that framerate being treated as drop-frame regardless of initialization parameters, #36  - depending on the version you are using and/or your expectations that may be the reason.

**Comment by sroucheray** [comment]
Sorry my explanation was not clear enough. I am using the latest version 1.3.3, and I saw the bug fix, but I tried to figure out how the 23.976 frame rate works.

This test fails:

```js
var t2 = new Timecode(3600 * 23.976, 23.976); 
expect(t2.toString()).to.be('01:00:00:00');
```

with this error

```
Error: expected '00:59:56:10' to equal '01:00:00:00'
```

but maybe this frame rate (neither integer nor drop frame) is a weirdo, this test passes: 

```js
var t2 = new Timecode(3600 * 24, 23.976); 
expect(t2.toString()).to.be('01:00:00:00');
```

Actually, the code might just work, but I must admit I have hard time understanding this specific frame rate case. Do you know any spec I can refer to (it seems not part of SMPTE) ?

**Comment by m1tk4** [comment]
This is actually expected behavior. In non-drop-frame timecodes like 23.976, the "wall clock" time will differ from your "timecode" time by 1/1000. All frames are counted 0-23, so your 25-th frame for example will have a timecode of 00:00:01:00, even though the time that has passed is actually (1/23.976)*24 == 1.001 seconds. NDF clock is "slower" than real clock by 1/1000. This is exactly the reason drop-frame was introduced.

See https://www.3playmedia.com/blog/drop-frame-non-drop-frame-affects-captions-subtitles/#:~:text=Drop%20frame%20(DF)%20timecode%20was,(once%20every%2033.33%20seconds). for another example.
